src/main/java/org/takes/facets/auth/social/PsFacebook.java

```diff
@@ -71,19 +84,44 @@
      * @param fkey Facebook key
      */
     public PsFacebook(final String fapp, final String fkey) {
+        this(
+            new JdkRequest(
+                new Href("https://graph.facebook.com/oauth/access_token")
+                    .with("client_id", fapp)
+                    .with("client_secret", fkey)
+                    .toString()
+            ),
+            new DefaultWebRequestor(),
+            fapp,
+            fkey
+        );
+    }
+
+    /**
+     * Ctor with proper requestor for testing purposes.
+     * @param frequest HTTP request for getting key
+     * @param frequestor Facebook response
+     * @param fapp Facebook app
+     * @param fkey Facebook key
+     * @checkstyle ParameterNumberCheck (3 lines)
+     */
+    public PsFacebook(final com.jcabi.http.Request frequest,
```

Comment:
this constructor can be package protected if it supposed to be used for test only